anker/tests/tests/amm.rs

```diff
@@ -0,0 +1,34 @@
+use lido::token::Lamports;
+use solana_program::program_pack::Pack;
+use solana_program_test::tokio;
+use solana_sdk::signer::Signer;
+use testlib::anker_context::Context;
+
+#[tokio::test]
+async fn test_successful_token_swap() {
+    let mut context = Context::new_with_initialized_token_pool().await;
+    let (st_sol_kp, st_sol_token) = context
+        .solido_context
+        .deposit(Lamports(10_000_000_000))
+        .await;
+
+    let ust_address = context
+        .solido_context
+        .create_spl_token_account(
+            context.token_pool_context.ust_mint_address,
+            st_sol_kp.pubkey(),
+        )
+        .await;
+
+    context
+        .swap_st_sol_for_ust(&st_sol_token, &ust_address, &st_sol_kp, 1_000_000_000, 0)
+        .await;
+
+    let ust_account = context.solido_context.get_account(ust_address).await;
+    let ust_spl_account: spl_token::state::Account =
+        spl_token::state::Account::unpack_from_slice(ust_account.data.as_slice()).unwrap();
```

Comment:
For stSOL and bSOL we have a helper method on the context to get the balance of an account, would be nice to add one for UST as well; it removes some of the implementation detail noise from the test, and it can return a more strongly typed result.